### DISCUSSION TITLE
CDK-EKS: Default VPC for cdk's eks cluster is not consistent with eksctl

### DISCUSSION THREAD

**Discussion by cseverino789** [aws/aws-cdk#14223] [General]
I've been working on a PoC for deploying an EKS Cluster with CDK and noticed a difference with working with eksctl.

The VPC created by eks.Cluster with CDK appears to be creating Elastic IP addresses and NAT Gateways per Public Subnet. Whereas, the eksctl creates a single Elastic IP and NAT Gateway and uses a single "PublicRouteTable" to handle routing to the 3 Public Subnets.

Ideally, I would expect the same default behavior? It seems eksctl is more efficient as it limits the constraint with Elastic IPs. Maybe a feature request?

**Comment by christseng89** [comment]
Exactly, if there is an cdk option for eksctl when create eks cluster, then we could replace eksctl to cdk directly.  I.e. created cloudformation resources same as eksctl, such as VPC, Security groups etc.  

Alternatively, using eksctl to create cluster and nodegroups, then using cdk to manipulate the rest of the changes required.   I.e. eksctl and cdk are complementary solutions to work together.  No more manual interventions.
